calblueprint/wecaresolar
# Video page (v2)

## 🖋 PR Description

Created the Video Page, containing the Video, Download Button, and Viewing Labels. If the video is cached, it will pull from IndexedDb, otherwise, it should display as an embedded video as usual.

- uses idb-keyval package (wraps IndexedDb)
- uses React Player to play videos
- missing styling for now
- the getId() function that uses regex was pulled from stackOverflow
- platform code (distinguishes between ios / android / unknown) was pulled from stack overflow, haven't been able to get it to work on android and test it from there yet though
- Kevin mentioned in the previous PR about getting the details of the error logs, but it might not be possible because of the idb-keyval package

### 🤳Screenshots (if applicable)

<img width="688" alt="Screen Shot 2020-11-21 at 6 55 13 PM" src="https://user-images.githubusercontent.com/46883811/99892601-38050100-2c2b-11eb-9a10-f812aac9334a.png">
<img width="678" alt="Screen Shot 2020-11-21 at 6 55 23 PM" src="https://user-images.githubusercontent.com/46883811/99892604-3c311e80-2c2b-11eb-9098-3598b8e30be2.png"> 



### 🐞 Bug Report

If you ran into any bugs, explain how you solved them


### 🧪 Testing

How do we test your feature?
Run npm start > Inspect Page > Applications > IndexedDB (you'll be able to see it added into the Database!)




cc: @joelenelatief
